lib/step.rb

```diff
@@ -114,17 +114,24 @@ def puts_full_output
     end
 
     def annotation_location(name)
-      formula = Formulary.factory(name)
+      formula, formula_path = begin
+        f = Formulary.factory(name)
+        [f, f.path]
+      rescue FormulaUnavailableError
+        [nil, @repository.glob("**/#{name}*").first]
+      end
+
       method_sym = command.second.to_sym
-      method_location = formula.method(method_sym).source_location if formula.respond_to?(method_sym)
 
-      if method_location.present? && (method_location.first == formula.path.to_s)
-        method_location
-      else
-        [formula.path, nil]
+      formula_line = if output? && (line = output.lines.find { |l| l.start_with? formula_path })
+        line.split(":").second
```

Comment:
What happens if there's not a `:` here?

Reply:
This is `nil`.

Reply:
Do we want to handle that case with the e.g. `elsif`?

Reply:
Yea, probably. Let me fix it.

Reply:
I've made the match stricter so that matches so that they are much more likely to pick up lines in a stacktrace and we're guaranteed to pick up a number in `line.split(":").second`.

Still working on `Formulary.path` though.

Reply:
How about instead of making this an `if`/`elsif` is you use a `||=` for the `elsif` so it's set if this `split` fails?

Reply:
I just reread the nonsense I just wrote -- apologies for the confusion.

The `split` will now never fail, since I specifically look for lines that look like `/#{formula_path}:\d+:/` (appropriately escaped).

What could happen is that I still pick up a number that's garbage (e.g. a line after the last line in the formula), so I suppose we could do some minimal validation after this.

Not sure it's worth it, though

Reply:
Yeh, agreed, not worth it. Happy to ship as-is.